frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx

```diff
@@ -53,13 +57,43 @@ const LogsTabForPodName: React.FC<{ podName: string }> = ({ podName }) => {
   const [downloading, setDownloading] = React.useState(false);
   const [downloadError, setDownloadError] = React.useState<Error | undefined>();
   const { scrollToBottom, onMount, editorOptions } = useCodeEditorAsLogs();
+  const [isDownloadDropdownOpen, setIsDownloadDropdownOpen] = React.useState(false);
 
   React.useEffect(() => {
     if (!isPaused && logs) {
       scrollToBottom();
     }
   }, [isPaused, logs, scrollToBottom]);
 
+  const allContainerlogs = (): void => {
+    let allContainerlogs = '';
+    const promises = podContainers.map(async function (podContainer) {
+      const logsIndividualContainer = await fetchContainerLogs(
+        namespace,
+        podName,
+        podContainer?.name,
+      );
+      return { stepName: podContainer.name, logsPerStep: logsIndividualContainer };
+    });
+    Promise.all(promises)
+      .then((results) => {
+        results.forEach(function (result) {
+          allContainerlogs =
+            allContainerlogs +
+            '-------------\n' +
+            result.stepName +
+            '\n-------------\n' +
+            result.logsPerStep;
+        });
+      })
+      .catch((error: string) => {
+        throw new Error(error);
+      })
+      .finally(() => {
+        downloadString(`${podName}.log`, allContainerlogs);
+      });
+  };
```

Comment:
See my root level review comment for more context on why this doesn't belong in the component. But let me also point out some things as well so you can move & improve.

* Promise.all(promises).catch => throw
    * Where are you throwing this error? Who is it for?
    * Perhaps you want to use `useNotification` -- can you render something inline?
* function name `allContainerLogs` -- what does this function name mean?
    * When you move this to another file, it'll showcase this is even more confusing name
    * The other download function is called `onDownload` -- it's an action... "on download, do x" -- `onDownloadAll` might be a better name for the onClick handler of the button
* When you download all... you don't lock up the UI, so I can spam this button I suspect; look at `onDownload`, see that it deals with state and throws the action effort to downloading to the utils
    * Consider the ownership here... 
       Component => disable states, rendering data, acting on user input
       Utility => download logic, if it needs to fetch, it can do that
    * If you use promises, this allows your component to say "utility, I expect you to promise to tell me when you're done" and then the component can act on that response, good (then) or bad (catch)
* This is more curious -- what was the thinking here? `.finally` => download anyway? -- We just errored and you threw something, you effectively broke the thread but promises catch this for you, what are you looking to download? Whatever you might or might not have? Why?
* `allContainerLogs`, you're making use of JS Closures to build a string... but this is async code, this is typically not a great idea... I think this basically suffers from imperative vs declarative thinking... let me explain
    * you have a goal, you set out to solve it flat => imperative
        * I need a string to hold stuff
        * loop and map some promises, async/await the values and make use of it, flattening promises
        * loop over the text and build out that top string
        * throw error if needed
        * download log at the end
    * you have a goal, and you describe steps to solve it => declarative
        * Fetch logs for each container
        * then, map the log and name into an object
        * then, reduce use that data to combine the content into a single file content string
        * then, download that result
        * catch, if something goes wrong, let the user know